packages/vite/src/node/server/middlewares/static.ts

```diff
@@ -72,9 +73,15 @@ export function serveStaticMiddleware(
   }
 }
 
-export function serveRawFsMiddleware(): Connect.NextHandleFunction {
+export function serveRawFsMiddleware(
+  config: ResolvedConfig
+): Connect.NextHandleFunction {
   const isWin = os.platform() === 'win32'
   const serveFromRoot = sirv('/', sirvOptions)
+  const serveRoot = path.resolve(
+    config.root,
+    config.server?.fsServeRoot || searchForWorkspaceRoot(config.root, 5)
```

Comment:
Should we set a maximum depth for searching up? Set `5` as a placeholder, but not sure about the number.

Reply:
Do you think that there will be performance issues if we avoid the limit? Or is this more about security?

Reply:
Yeah I am thinking about security. For example, I can create a `pnpm-workspace.yaml` under root to fool vite exposing the entire filesystem (or just without users awareness). Having the depth limit can't really resolve the problem completely, but I haven't came out with a better way.

Reply:
Maybe we could remove the limit but bail out if we reached root? I don't think people are going to have a monorepo at root level, no? If we use 5, at least in my system you will always hit root anyways.

Reply:
Alright, removed.